src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,304 @@
+            args: ["{that}", "{that}.model.appId", "{gpii.devpmt.editPrefs}.model.contextNames", "{gpii.devpmt.editPrefs}.editProductEnabled"]
+        }
+    }
+});
+
+/**
+ * Clicking Add on the Product Dialog adds this product to the preference set.
+ * A preference safe must have at least one context (prefset) in order to add
+ * a product to it.
+ *
+ * @param  {Object} that               Dialog instance
+ * @param  {String} appId              String indicating the appId
+ *                                     ex. http://registry.gpii.net/applications/com.android.freespeech
+ * @param  {Array} contextNames        Array of Strings with the name/id the context is
+ *                                     keyed by. The product will be initially added to the first
+ *                                     item in this array.
+ * @param  {Function} editProductEnabled Invoker from `gpii.devpmt.editPrefs` to add the
+ *                                       product to.
+ */
+gpii.devpmt.dialogs.confirmAddProductDialog.acceptConfirmDialog = function (that, appId, contextNames, editProductEnabled) {
+    // In this case we actually need to close the dialog first... as the page
+    // rerenders based on a model listener when the product is enabled, and does
+    // wonky things... such as removing the ability to vertical scroll. Should look
+    // at reworking this perhaps.
+    that.closeDialog();
+    // TODO Ontology!!!
+    var appUrl = "http://registry.gpii.net/applications/" + appId;
+    if (contextNames.length > 0) {
+        editProductEnabled(true, contextNames[0],  appUrl);
+    }
+};
+
+/**
+ * Confirm NP Set Save Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmSaveDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-confirmSave-dialog"
+    },
+    model: {
+        unsavedChanges: []
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.confirmSaveDialog.acceptConfirmDialog",
+            args: ["{that}", "{gpii.devpmt.editPrefs}.savePrefset"]
+        }
+    }
+});
+
+gpii.devpmt.dialogs.confirmSaveDialog.acceptConfirmDialog = function (that, saveFunc) {
+    saveFunc();
+    that.closeDialog();
+};
+
+
+/**
+ * Confirm Removing Product Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmRemoveProductDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-confirmRemoveProduct-dialog"
+    },
+    model: {
+        appId: "",
+        name: "",
+        product: "",
+        context: ""
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.confirmRemoveProductDialog.acceptConfirmDialog",
+            args: ["{that}", "{gpii.devpmt.editPrefs}"]
+        }
+    }
+});
+
+gpii.devpmt.dialogs.confirmRemoveProductDialog.acceptConfirmDialog = function (that, devpmt) {
```

Comment:
Be more specific with input parameters rather than passing the entire "{that}" or "{devpmt}".

Reply:
I've taken a fair pass at scoping things in this package more.